pkg/server/endpoints/entryfetcher_test.go

```diff
@@ -215,6 +232,49 @@ func TestRunRebuildCacheTask(t *testing.T) {
 	sendResult(req, entryMap, nil)
 }
 
+func TestRunPruneEventsTask(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	watchErr := make(chan error, 1)
+	defer func() {
+		cancel()
+		select {
+		case err := <-watchErr:
+			assert.NoError(t, err)
+		case <-time.After(5 * time.Second):
```

Comment:
you can relay on ctx done here

Reply:
Are you requesting to put `case <-ctx.Done():`?

Reply:
since context will finish (with timeout) you can rely ctx.Done()

Reply:
In this case we are calling `cancel()` just a few lines above on line 239. If we use `ctx.Done()` that will always be true so we never check for the watchErr.